packages/contentstack-import/src/utils/content-type-helper.ts

```diff
@@ -138,7 +146,6 @@ export const removeReferenceFields = async function (
     ) {
       flag.supressed = true;
       schema[i].reference_to = ['sys_assets'];
-      return true;
```

Comment:
@shafeeqd959  / @aman19K  Could you confirm this change as well.?

cc @netrajpatel @abhinav-from-contentstack @cs-raj

Reply:
In `src/import/modules-js/global-field.js` file, removeReferenceFields sets value for `flag.supressed` property and based on the flag value we decide if global-field needs to be pushed to _globalField_pending.
https://github.com/contentstack/cli/blob/c005a1343aad6a16d041dc68327e52f1ee6df5e6/packages/contentstack-import/src/import/modules-js/global-fields.js#L66C10-L71C12

For the corresponding ts file, the dependence on `flag.supressed` has been removed. And the value returned by removeReferenceFields is being used to determine if global field needs to be pushed to pending array.
https://github.com/contentstack/cli/blob/c005a1343aad6a16d041dc68327e52f1ee6df5e6/packages/contentstack-import/src/import/modules/global-fields.ts#L119-L122

I think, this issue can be solved by re-introducing the flag in ts file again
https://github.com/contentstack/cli/blob/c005a1343aad6a16d041dc68327e52f1ee6df5e6/packages/contentstack-import/src/import/modules-js/global-fields.js#L63

cc: @shafeeqd959 @antonyagustine @aman19K

Reply:
@shafeeqd959  Kindly validate the accuracy of the identified observations and the proposed suggestion.

Reply:
@abhinav-from-contentstack why do we need to remove the return statement here, if we remove it that will break the rewrite/ts version of global field import. Please let me know

Reply:
@shafeeqd959 If there are many fields in the global field, such as json_rte, block, and group, and we reference CT in each field, json_rte will return right after json_rte and won't remove references to block and group. It appears that we are presuming a sequence of fields just from looking at the code. Correct me if I'm wrong.

cc: @abhinav-from-contentstack @antonyagustine

Reply:
This is correct, I had a discussion with @shafeeqd959 and we have come to the conclusion that re-introducing `flag.supressed` in the `ts` version is the way to fix this issue for now.